src/win/fs.c

```diff
@@ -627,6 +644,10 @@ void fs__write(uv_fs_t* req) {
   if (offset != -1) {
     memset(&overlapped, 0, sizeof overlapped);
     overlapped_ptr = &overlapped;
+    if (SetFilePointerEx(handle, zero_offset, &original_position,
+                         FILE_CURRENT)) {
+      restore_position = 1;
```

Comment:
As per [MSDN SetFilePointerEx doc](https://msdn.microsoft.com/pl-pl/library/windows/desktop/aa365542.aspx) this call can fail if the handle is not a file or if it was opened with `FILE_FLAG_NO_BUFFERING`. We test handle type and libuv does not uses that flag, so I guess we are safe.